docs/pages/api/index.md

```diff
@@ -12,33 +12,115 @@ env = gym.make('CartPole-v0')
 ```
 
 ## Interacting with the Environment
-This example will run an instance of `CartPole-v0` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the classic [cart-pole](https://www.youtube.com/watch?v=J7E6_my3CHk&ab_channel=TylerStreeter) problem
+Gym implements the classic "agent-environment loop":
+
+<img src="https://gym.openai.com/assets/docs/aeloop-138c89d44114492fd02822303e6b4b07213010bb14ca5856d2d49d6b62d88e53.svg" width=20%/>
+
+The agent performs some actions in the environment (usually by passing some control inputs to the environment, e.g. torque inputs of motors) and observes
+how the environment's state changes. One such action-observation exchange is referred to as a *timestep*. 
+
+The goal in RL is to manipulate the environment in some specific way. For instance, we want the agent to navigate a robot
+to a specific point in space. If it succeeds in doing this (or makes some progress towards that goal), it will receive a positive reward
+alongside the observation for this timestep. The reward may also be negative or 0, if the agent did not yet succeed (or did not make any progress). 
+The agent will then be trained to maximize the reward it accumulates over many timesteps.
+
+After some timesteps, the environment may enter a terminal state. For instance, the robot may have crashed! In that case,
+we want to reset the environment to a new initial state. The environment issues a done signal to the agent if it enters such a terminal state.
+Not all done signals must be triggered by a "catastrophic failure": Sometimes we also want to issue a done signal after
+a fixed number of timesteps, or if the agent has succeeded in completing some task in the environment.
+
+Let's see what the agent-environment loop looks like in Gym.
+This example will run an instance of `LunarLander-v2` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the environment
 
 ```python
 import gym
-env = gym.make('CartPole-v0')
-env.reset()
-for _ in range(1000): 
-	env.render()  # by default `mode="human"`(GUI), you can pass `mode="rgb_array"` to retrieve an image instead
-	env.step(env.action_space.sample())  # take a random action 	
+env = gym.make("LunarLander-v2")
+env.action_space.seed(42)
+
+observation, info = env.reset(seed=42, return_info=True)
+
+for _ in range(1000):
+    env.render()
+    observation, reward, done, info = env.step(env.action_space.sample())
+
+    if done:
+      observation, info = env.reset(return_info=True)
+
 env.close()
 ```
 
 The output should look something like this
 
-![cartpole-no-reset](https://user-images.githubusercontent.com/28860173/129241283-70069f7c-453d-4670-a226-854203bd8a1b.gif)
+![lunar_lander](https://user-images.githubusercontent.com/15806078/153222406-af5ce6f0-4696-4a24-a683-46ad4939170c.gif)
+
+
+## Commonly used methods
```

Comment:
I'd change this to maybe "Standard methods" or "Common API methods" or something like that

Basically we want to imply that this is a complete list of methods shared between **all** gym environments, not just that they happen to be used a lot